pkg/kubelet/status/status_manager.go

```diff
@@ -220,17 +220,12 @@ func (m *manager) SetPodStatus(pod *v1.Pod, status v1.PodStatus) {
 	m.updateStatusInternal(pod, status, pod.DeletionTimestamp != nil)
 }
 
-func (m *manager) SetContainerReadiness(podUID types.UID, containerID kubecontainer.ContainerID, ready bool) {
+func (m *manager) SetContainerReadiness(pod *v1.Pod, containerID kubecontainer.ContainerID, ready bool) {
 	m.podStatusesLock.Lock()
 	defer m.podStatusesLock.Unlock()
 
-	pod, ok := m.podManager.GetPodByUID(podUID)
-	if !ok {
-		klog.V(4).InfoS("Pod has been deleted, no need to update readiness", "podUID", string(podUID))
-		return
-	}
-
-	oldStatus, found := m.podStatuses[pod.UID]
+	podUID := pod.UID
```

Comment:
Do you really need this variable? It's used only once.

Reply:
removed.